lib/src/stream_queue.dart

```diff
@@ -330,23 +316,24 @@ class StreamQueue<T> {
   ///   }
   /// }
   /// ```
-  Future<bool> withTransaction(Future<bool> Function(StreamQueue<T>) callback) {
+  Future<bool> withTransaction(
+      Future<bool> Function(StreamQueue<T>) callback) async {
     var transaction = startTransaction();
 
-    /// Avoid async/await to ensure that [startTransaction] is called
-    /// synchronously and so ends up in the right place in the request queue.
     var queue = transaction.newQueue();
-    return callback(queue).then((result) {
-      if (result) {
-        transaction.commit(queue);
-      } else {
-        transaction.reject();
-      }
-      return result;
-    }, onError: (Object error) {
+    bool result;
+    try {
+      result = await callback(queue);
+    } on Object {
```

Comment:
I think its more common to do `catch (_)` if you want to catch all exceptions and ignore the error, although I don't see a problem with this I think its a bit harder to read. Made me do a double take lol.

Reply:
Done.